assets/js/blocks/price-filter/block.js

```diff
@@ -59,14 +95,42 @@ const PriceFilterBlock = ( { attributes, isEditor = false } ) => {
 	// Updates the query based on slider values.
 	const onSubmit = useCallback(
 		( newMinPrice, newMaxPrice ) => {
-			setMinPriceQuery(
-				newMinPrice === minConstraint ? undefined : newMinPrice
-			);
-			setMaxPriceQuery(
-				newMaxPrice === maxConstraint ? undefined : newMaxPrice
-			);
+			if ( filteringForPhpTemplate && window ) {
+				// For block templates that render the PHP Classic Template block we need to add the filters as params and reload the page.
+				const url = new URL( window.location );
+				const currentParams = new URLSearchParams( url.search );
```

Comment:
Perhaps here we could also use https://developer.wordpress.org/block-editor/reference-guides/packages/packages-url/#getqueryargs ?